crates/ra_lsp_server/src/conv.rs

```diff
@@ -266,12 +271,20 @@ impl TryConvWith for SourceChange {
                 })
             }
         };
-        let source_file_edits = self.source_file_edits.try_conv_with(world)?;
-        let file_system_edits = self.file_system_edits.try_conv_with(world)?;
+        let mut document_changes: Vec<DocumentChangeOperation> = Vec::new();
+        for resource_op in self.file_system_edits.try_conv_with(world)? {
+            document_changes.push(DocumentChangeOperation::Op(resource_op));
+        }
+        for text_document_edit in self.source_file_edits.try_conv_with(world)? {
+            document_changes.push(DocumentChangeOperation::Edit(text_document_edit));
```

Comment:
the above two `for` loop can be more rustic with iterator. 
lint: `.chain()` and `.map()`.

Reply:
Yeah, though I personally don't have a strong preference towards HoFs or for loops: for loops play much better if you have `?` in the body of the loop.